src/Activities/Internal/EventSource/ActivityEventSender.cs

```diff
@@ -6,25 +6,28 @@
 using System.Diagnostics;
 using System.Globalization;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using Microsoft.Omex.Extensions.Abstractions;
 using Microsoft.Omex.Extensions.Abstractions.Activities;
 using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
 using Microsoft.Omex.Extensions.Abstractions.ExecutionContext;
+using Microsoft.Omex.Extensions.Abstractions.Option;
 
 namespace Microsoft.Omex.Extensions.Activities
 {
 	internal sealed class ActivityEventSender : IActivitiesEventSender
 	{
-		public ActivityEventSender(ActivityEventSource eventSource, IExecutionContext executionContext, ILogger<ActivityEventSender> logger)
+		public ActivityEventSender(ActivityEventSource eventSource, IExecutionContext executionContext, ILogger<ActivityEventSender> logger, IOptions<MonitoringOption> options)
```

Comment:
Why not just remove the call to AddOmexActivitySource from the apps/services that are being updated?
https://github.com/microsoft/Omex/blob/32b76da25bcff08cb8ea7c6882371ce65ad92257/src/Activities/ServiceCollectionExtensions.cs#L25C36-L25C57

Reply:
Given that it is a dirty hack in general, an alternative approach, which pollutes less code would be to read IConfiguration directly in AddOmexServices/AddOmexActivitySource and decide whether to call AddOmexActivitySource or not.